### PR TITLE
Fix application facades

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Restore facades on application 16 that were incorrectly dropped and increment facade to 17

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

Verify unit tests pass

Verify DestroyUnits facade can be called using application facade 15 or 16

### CODE REVIEW

File: api/facadeversions.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ var facadeVersions = map[string]int{
 	"AllModelWatcher":              4,
 	"AllWatcher":                   3,
 	"Annotations":                  2,
-	"Application":                  16,
+	"Application":                  17,
```
Comment (jameinel) [inline api/facadeversions.go:21]: Thinking about this line change, I feel like now that we have actually gone to 3.0 and have dropped support for old versions, all of these single integers actually need to become tuples with start/end, or lists with explicit numbers.

I'm happiest with tuples, and also very happy to not introduce this in 3.0, but make sure we start getting it right moving forward. (We sort of cheated heavily by just rejecting any target that is <3, but we *should* support a range of facade versions)

**Review by SimonRichardson — COMMENTED** [review]
Can we get a test for v17, ensuring that it no longer exists.

File: api/facadeversions.go (hunk 1 above)
Comment (SimonRichardson) [reply api/facadeversions.go:21]: Agreed.

**Review by SimonRichardson — APPROVED** [review]

**Pull request merged**